### PR TITLE
develop: changes required to process global NetCDF files.

### PR DESCRIPTION
**Pull request by JeffBeck-NOAA**
## DESCRIPTION OF CHANGES: 
Changes to allow selection and pre-processing of global NetCDF files.

## TESTS CONDUCTED: 
@fentoad72 tested end-to-end simulation using global NetCDF data as ICs/LBCs.

## ISSUE (optional): 
Addresses issue #401.

## CONTRIBUTORS (optional): 
Thank you for testing, @fentoad72.

### CODE REVIEW

**Review by fentoad72 — APPROVED** [review]
Changes to include netcdf input.  Approved.

File: scripts/exregional_make_ics.sh [hunk 1]
```diff
@@ -192,13 +192,13 @@ esac
 # Name of the external model from which we are obtaining the fields
 # needed to generate the ICs.
 #
-# fn_atm_nemsio:
-# Name (not including path) of the nemsio file generated by the external
-# model that contains the atmospheric fields.
+# fn_atm:
```
Comment (gsketefian) [inline scripts/exregional_make_ics.sh:195]: Are these variables (fn_atm and fn_sfc) used only when the external model is FV3GFS or GSMGFS?  If so, can you state that here, and maybe change the the names of these variables to (fv3)gfs_atm_fn and (fv3)gfs_sfc_fn to emphasize they only apply to FV3GFS and GSMGFS?

**Review by fentoad72 — APPROVED** [review]
Changes made to include netCDF input.  Approved

File: scripts/exregional_make_lbcs.sh [hunk 2]
```diff
@@ -192,13 +192,9 @@ esac
...
 # needed to generate the LBCs.
 #
-# fn_atm_nemsio:
-# Name (not including path) of the nemsio file generated by the external
-# model that contains the atmospheric fields.
-#
-# fn_sfc_nemsio:
-# Name (not including path) of the nemsio file generated by the external
-# model that contains the surface fields.
+# fn_atm:
```
Comment (gsketefian) [inline scripts/exregional_make_lbcs.sh:195]: Same comment/question as in exregional_make_ics.sh file regarding this variable.

**Comment by gsketefian** [comment]
In the PR message, can you say what tests were conducted and what the results were?

File: scripts/exregional_make_ics.sh (hunk 1 above)
Comment (JeffBeck-NOAA) [reply scripts/exregional_make_ics.sh:195]: Currently, fn_atm and fn_sfc are only applicable for FV3GFS or GSMGFS netcdf or nemsio files, but there will likely be support for other netcdf external model files in the near future, so I'd be hesitant to make the variables model-name specific.  I will add a comment specifying the support external model types for each category.

File: scripts/exregional_make_lbcs.sh (hunk 2 above)
Comment (JeffBeck-NOAA) [reply scripts/exregional_make_lbcs.sh:195]: Added comments in the latest commit.

**Comment by JeffBeck-NOAA** [comment]
> In the PR message, can you say which WE2E tests were conducted and what the results were? If they were new tests with netcdf files, we should put them in the WE2E test system so we can test this feature in the future.

@fentoad72, since you tested this functionality, can you respond to Gerard's question?  Thanks!

File: ush/get_extrn_mdl_file_dir_info.sh [hunk 3]
```diff
@@ -375,6 +375,29 @@ fi
 #        fns=( "gfs.t${hh}z.pgrb2.0p25.f000" )  # Get only 0.25 degree files for now.
         fns_on_disk=( "gfs.t${hh}z.pgrb2.0p25.f000" )  # Get only 0.25 degree files for now.
         fns_in_arcv=( "gfs.t${hh}z.pgrb2.0p25.f000" )  # Get only 0.25 degree files for now.
+
```
Comment (gsketefian) [inline ush/get_extrn_mdl_file_dir_info.sh:378]: Can you make this section identical to what's in PR #427 (i.e. set things to empty strings and print out a message)?

File: ush/get_extrn_mdl_file_dir_info.sh [hunk 4]
```diff
@@ -471,6 +494,27 @@ and analysis or forecast (anl_or_fcst):
         fns_on_disk=( "${fcst_hhh[@]/#/$prefix}" )
         fns_in_arcv=( "${fcst_hhh[@]/#/$prefix}" )
 
+
```
Comment (gsketefian) [inline ush/get_extrn_mdl_file_dir_info.sh:512]: Can you make this section identical to what's in PR #427 (i.e. set things to empty strings and print out a message)?

File: ush/get_extrn_mdl_file_dir_info.sh [hunk 5]
```diff
@@ -816,6 +860,27 @@ has not been specified for this external model:
 
       arcv_fns="${arcv_fns}gfs_pgrb2"
 
+    elif [ "${fv3gfs_file_fmt}" = "netcdf" ]; then
+
+#
```
Comment (gsketefian) [inline ush/get_extrn_mdl_file_dir_info.sh]: Can you make this section identical to what's in PR #427 (i.e. set things to empty strings and print out a message)?

**Review by gsketefian — APPROVED** [review]

**Pull request merged**